### PR TITLE
fix(dapp): unhandled error when adding token in Dapp

### PR DESCRIPTION
**Pull request by jasonzhouu**

### CODE REVIEW

File: packages/dapp/src/components/TokenDapp.tsx [hunk 1]
```diff
@@ -25,6 +25,9 @@ export const TokenDapp: FC = () => {
   const [transactionStatus, setTransactionStatus] = useState<
     "idle" | "approve" | "pending" | "success"
   >("idle")
+  const [errorMessageOfAddERC20Token, setErrorMessageOfAddERC20Token] =
```
Comment (delaaxe) [inline packages/dapp/src/components/TokenDapp.tsx:28]:
```suggestion
  const [addTokenError, setAddTokenError] =
```

File: packages/dapp/src/components/TokenDapp.tsx [hunk 2]
```diff
@@ -25,6 +25,9 @@ export const TokenDapp: FC = () => {
...
+    useState("")
+  const [errorMessageOfAddETHToken, setErrorMessageOfAddETHToken] = useState("")
```
Comment (delaaxe) [inline packages/dapp/src/components/TokenDapp.tsx:30]: One error message variable should be enough

File: packages/dapp/src/components/TokenDapp.tsx [hunk 3]
```diff
@@ -207,7 +210,15 @@ export const TokenDapp: FC = () => {
...
-          onClick={() => addToken(tokenAddress)}
+          onClick={() => {
+            addToken(tokenAddress)
+              .then(() => {
+                setErrorMessageOfAddERC20Token("")
+              })
+              .catch((error) => {
+                setErrorMessageOfAddERC20Token(error.message)
+              })
+          }}
```
Comment (delaaxe) [inline packages/dapp/src/components/TokenDapp.tsx:221]: Please prefer async/await notation

File: packages/dapp/src/components/TokenDapp.tsx [hunk 4]
```diff
@@ -221,14 +232,23 @@ export const TokenDapp: FC = () => {
...
             style={{ marginLeft: ".6em" }}
-            onClick={() => addToken(ethAddress)}
+            onClick={() => {
+              addToken(ethAddress)
+                .then(() => {
+                  setErrorMessageOfAddETHToken("")
+                })
+                .catch((error) => {
+                  setErrorMessageOfAddETHToken(error.message)
+                })
```
Comment (delaaxe) [inline packages/dapp/src/components/TokenDapp.tsx:250]: Please prefer async/await notation

**Review by delaaxe — CHANGES_REQUESTED** [review]

File: packages/dapp/src/components/TokenDapp.tsx (hunk 2 above)
Comment (jasonzhouu) [reply packages/dapp/src/components/TokenDapp.tsx]: If only one variable is used, an error occurs when adding the erc20 token, the error message will be displayed in both places

File: packages/dapp/src/components/TokenDapp.tsx (hunk 3 above)
Comment (jasonzhouu) [reply packages/dapp/src/components/TokenDapp.tsx]: OK. updated

File: packages/dapp/src/components/TokenDapp.tsx (hunk 4 above)
Comment (jasonzhouu) [reply packages/dapp/src/components/TokenDapp.tsx]: updated

File: packages/dapp/src/components/TokenDapp.tsx (hunk 2 above)
Comment (delaaxe) [reply packages/dapp/src/components/TokenDapp.tsx]: It's fine displaying just one error in one place

**Review by delaaxe — APPROVED** [review]

**Comment by jasonzhouu** [comment]
Thank you

**Pull request merged**